### ISSUE TITLE
Pass parameters to flows

### ISSUE THREAD

**Issue by naicigam** [SalesforceFoundation/CumulusCI#315]
I am trying to run the `ci_beta` flow, but trying to pass a specific package version to the `install_managed_beta` task.

I tried the `-o ` parameter, but does not work at a workflow level, so the option that comes to my mind is running task by task and pass the version using the `cci task run install_managed_beta -o version 1.1b` command line.

Is there another way of doing this that allows me to take advantage of future updates to the flows?

Thanks!
Juan M.

**Comment by jlantz** [comment]
That's definitely an enhancement to flows that I'd like to see, but at the moment passing options to flows isn't possible.  The ci_beta flow looks up the latest release from Github Releases for the repo and installs that version.

The code for flows is actually pretty simple so I think this wouldn't be too hard to add:
https://github.com/SalesforceFoundation/CumulusCI/blob/master/cumulusci/core/flows.py

**Comment by cdcarter** [comment]
It's relatively quick to override a single flow task definition in local YAML:

```
    ci_beta:
        tasks:
            4:
                task: install_managed_beta
                options:
                    version: 1.1b
```

This isn't ideal, but its a quick git stash pop...

**Comment by jlantz** [comment]
@cdcarter nice workaround that I wouldn't have thought of ;)

The local cumulusci.yml lives under ~/.cumulusci/YOUR_PROJECT_NAME/cumulusci.yml

It's not there by default but if you create it, it will be merged on top of the repo's cumulusci.yml in your local environment.

**Comment by jlantz** [comment]
This is now included in 2.0.0-beta36 and above

**Comment by initdotd** [comment]
I know this is very old post, but had a hard time finding a way to get passing args to the flow that would then be passed to the task.  This is the basic syntax as I understand it.  It does work for me.

```
cci flow run some_flow --org theOrg -o task__param value
```

This is an example line taken from a manually triggered GitHub action where the action prompts the user for the package version id (04t) and then passes it to the cci flow run command.  The command then notifies the flow of the additional options specific to the "promote_package_version" task with an argument named "version_id".  NOTE: There is a double underscore in between the task and arg names.  The last argument is the value provided as input when triggering the flow manually.

```
cci flow run release_unlocked_production --org release -o promote_package_version__version_id ${{ github.event.inputs.version }}
```

I hope this helps someone else.  Happy coding!
